src/effects/builtin/whitenoiseeffect.cpp

```diff
@@ -0,0 +1,76 @@
+#include "effects/builtin/whitenoiseeffect.h"
+
+#include "util/rampingvalue.h"
+
+namespace {
+const QString dryWetParameterId = QStringLiteral("dry_wet");
+} // anonymous namespace
+
+// static
+QString WhiteNoiseEffect::getId() {
+    return QStringLiteral("org.mixxx.effects.whitenoise");
+}
+
+// static
+EffectManifestPointer WhiteNoiseEffect::getManifest() {
+    EffectManifestPointer pManifest(new EffectManifest());
+    pManifest->setId(getId());
+    pManifest->setName(QObject::tr("White Noise"));
+    pManifest->setAuthor("The Mixxx Team");
+    pManifest->setVersion("1.0");
+    pManifest->setDescription(QObject::tr("Creates active white noise sound"));
+
+    // This is dry/wet parameter
+    EffectManifestParameterPointer intensity = pManifest->addParameter();
+    intensity->setId(dryWetParameterId);
+    intensity->setName(QObject::tr("Dry/Wet"));
+    intensity->setDescription(QObject::tr("Set the dry wet parameter which makes the noise louder or more quiet"));
+    intensity->setControlHint(EffectManifestParameter::ControlHint::KNOB_LOGARITHMIC);
+    intensity->setSemanticHint(EffectManifestParameter::SemanticHint::UNKNOWN);
+    intensity->setUnitsHint(EffectManifestParameter::UnitsHint::UNKNOWN);
+    intensity->setDefaultLinkType(EffectManifestParameter::LinkType::LINKED);
+    intensity->setMinimum(0);
+    intensity->setDefault(0);
+    intensity->setMaximum(1);
+
+    return pManifest;
+}
+
+WhiteNoiseEffect::WhiteNoiseEffect(EngineEffect* pEffect)
+        : m_pDryWetParameter(pEffect->getParameterById(dryWetParameterId)) {
+}
+
+WhiteNoiseEffect::~WhiteNoiseEffect() {
+}
+
+void WhiteNoiseEffect::processChannel(
+        const ChannelHandle& handle,
+        WhiteNoiseGroupState* pGroupState,
+        const CSAMPLE* pInput,
+        CSAMPLE* pOutput,
+        const mixxx::EngineParameters& bufferParameters,
+        const EffectEnableState enableState,
+        const GroupFeatureState& groupFeatures) {
+    Q_UNUSED(handle);
+    Q_UNUSED(groupFeatures);
+    Q_UNUSED(enableState);
+
+    WhiteNoiseGroupState& gs = *pGroupState;
+    
+    CSAMPLE drywet = m_pDryWetParameter->value();
+    RampingValue<CSAMPLE_GAIN> drywet_ramping_value(
+            drywet, gs.prev_drywet, bufferParameters.framesPerBuffer());
+
+    std::uniform_real_distribution<> r_distributor(0.0, 1.0);
+
+    for (unsigned int i = 0; i < bufferParameters.samplesPerBuffer(); i ++) {
+        CSAMPLE_GAIN drywet_ramped = drywet_ramping_value.getNext();
+
+        float noise = static_cast<float>(
+                r_distributor(gs.gen));
+
+        pOutput[i] = pInput[i] * (1 - drywet_ramped) + noise * drywet_ramped;
+    }
+
+    gs.prev_drywet = drywet;
```

Comment:
```suggestion
if (enableState == EffectEnableState::Disabling) {
    gs.prev_drywet = 0;
} else {
    gs.prev_drywet = drywet;
}
```
This makes the effect ramp from dry when it is toggled on and off.